src/mod/server.mod/servmsg.c

```diff
@@ -1279,6 +1280,28 @@ static void server_resolve_failure(int servidx)
   lostdcc(servidx);
 }
 
+static void do_oident() {
+  char *home = getenv("HOME");
+  char path[121], buf[(sizeof "global{reply \"\"}") + USERLEN];
+  int nbytes;
+  int fd;
+
+  if (!home) {
+    putlog(LOG_SERV, "*",
+           "do_oident(): getenv(): variable HOME is not in the current environment");
+    return;
+  }
+  snprintf(path, sizeof path, "%s/.oidentd.conf", home);
```

Comment:
I suggest checking the return value of `snprintf` here; otherwise, if the path doesn't fit into the buffer, eggdrop could write to the wrong file.

Reply:
Thank you. done.